spritz/spritz-android/src/main/java/com/novoda/spritz/Spritz.java

```diff
@@ -0,0 +1,257 @@
+                .setDuration(defaultSwipeAnimationDuration);
+        finishSwipeAnimation.setInterpolator(getSwipeBackwardsInterpolatorFor(currentPage));
+        finishSwipeAnimation.addUpdateListener(defaultAnimatorUpdateListener());
+
+        return Collections.<Animator>singletonList(finishSwipeAnimation);
+    }
+
+    private float getAutoPlayEndProgressForPosition(int position) {
+        return ((float) spritzPagesWithOffset.get(position).autoPlayEnd()) / totalAnimationDuration;
+    }
+
+    private TimeInterpolator getSwipeBackwardsInterpolatorFor(SpritzPageWithOffset currentPage) {
+        TimeInterpolator pageSwipeBackwardsInterpolator = currentPage.swipeBackwardsInterpolator();
+        if (pageSwipeBackwardsInterpolator != null) {
+            return pageSwipeBackwardsInterpolator;
+        }
+        return this.defaultSwipeBackwardsInterpolator;
+    }
+
+    private ValueAnimator.AnimatorUpdateListener defaultAnimatorUpdateListener() {
+        return new ValueAnimator.AnimatorUpdateListener() {
+            @Override
+            public void onAnimationUpdate(ValueAnimator animation) {
+                lottieAnimationView.setProgress((Float) animation.getAnimatedValue());
+            }
+        };
+    }
+
+    public static class Builder {
+
+        private static final long DEFAULT_SWIPE_ANIMATION_DURATION = TimeUnit.MILLISECONDS.toMillis(500);
+
+        private final LottieAnimationView lottieAnimationView;
+
+        private long defaultSwipeAnimationDuration = DEFAULT_SWIPE_ANIMATION_DURATION;
+        private TimeInterpolator defaultSwipeForwardInterpolator = new LinearInterpolator();
+        private TimeInterpolator defaultSwipeBackwardsInterpolator = new LinearInterpolator();
+        private List<SpritzPageWithOffset> spritzPagesWithOffset;
+
+        private Builder(LottieAnimationView lottieAnimationView) {
+            this.lottieAnimationView = lottieAnimationView;
+            spritzPagesWithOffset = new ArrayList<>();
+        }
+
+        public Builder withDefaultSwipeAnimationDuration(long defaultSwipeAnimationDuration, TimeUnit timeUnit) {
+            this.defaultSwipeAnimationDuration = timeUnit.toMillis(defaultSwipeAnimationDuration);
+            return this;
+        }
+
+        public Builder withDefaultSwipeForwardInterpolator(TimeInterpolator swipeForwardInterpolator) {
+            this.defaultSwipeForwardInterpolator = swipeForwardInterpolator;
+            return this;
+        }
+
+        public Builder withDefaultSwipeBackwardsInterpolator(TimeInterpolator swipeBackwardsInterpolator) {
+            this.defaultSwipeBackwardsInterpolator = swipeBackwardsInterpolator;
+            return this;
+        }
+
+        public Builder withPages(SpritzPage... spritzPages) {
+            this.spritzPagesWithOffset = SpritzPageWithOffset.fromSpritzPages(spritzPages);
+            return this;
+        }
+
+        public Spritz attachTo(ViewPager viewPager) {
```

Comment:
I'm not a fan of this, it seems like we're trying to emphasise on fluent API missing flexibility and explicitness. I'd expect a `Spritz` instance to implement `ViewPager.OnPageChangeListener`  and let the final user decide when to attach or *detach* it. (The latter case is not considered at all)

Reply:
I'm actually ok with this being the point where you attach to a viewPager, but then we need a `detach()` method on `Spritz` itself as Toto suggests.

Reply:
Sorry I wasn't clear: I meant I'm not a fan to be built-in the builder. It should be part of the `Spritz` instance API imo.

Reply:
Yeah I understood that, I meant I don't have a problem with that to be honest

Reply:
So @mr-archano what would be your idea for this API to be?

Reply:
I think what @mr-archano means is removing `attachTo` from the builder API and add it to the Spritz object returned by the builder API.
So you'll do:
```
Spritz sp = Spritz
    .with(lottieAnimationView)
    .withDefaultSwipeAnimationDuration(300, TimeUnit.MILLISECONDS)
    .withDefaultSwipeForwardInterpolator(SWIPE_FORWARD_INTERPOLATOR)
    .withSteps(...)

// Later on
sp.attachTo(viewPager);
```

Reply:
Or maybe he meant returning a `ViewPager.OnPageChangeListener` and let the user setting it on the `ViewPager`?